src/main/java/com/google/sps/servlets/EventServlet.java

```diff
@@ -65,7 +76,89 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
     @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        // ToDo: Mingyi implment this method.
-        response.getWriter().println("You've called the post request");
+        String jsonBody = new BufferedReader(new InputStreamReader(request.getInputStream())).lines().collect(
+            Collectors.joining("\n"));
+
+        // Error handling: return error if jsonBody is empty
+        if (jsonBody == null || jsonBody.trim().length() == 0) {
+            System.out.println("the String fails to be parsed");
+            throw new NullPointerException();
+        }
+
+        // Parse the jsonBody String to a Gson object
+        Gson gson = new Gson();
+        Type type = new TypeToken<Map<String, String>>(){}.getType();
+        Map<String,String> eventJsonAsMap = gson.fromJson(jsonBody, type);
+
+        // Check if all the fields are present
+        if (!checkAllEventFieldsPresent(eventJsonAsMap)){
+            // throw error
+            System.out.println("Not all the fields in the map are present");
+            throw new NullPointerException();
+        }
+
+        // Get the value entered in the form  
+        Double event_Lat_D = Double.parseDouble(eventJsonAsMap.get("lat"));
+        Double event_Lng_D = Double.parseDouble(eventJsonAsMap.get("lng"));
+        String event_name = eventJsonAsMap.get("name");
+        String event_desc = eventJsonAsMap.get("description");
+        String event_loc = eventJsonAsMap.get("loc");
+        TimestampValue event_time = TimestampValue.of(Timestamp.parseTimestamp((eventJsonAsMap.get("date"))));
+        String event_type = eventJsonAsMap.get("type");
+        String event_sub = eventJsonAsMap.get("subject");
+        if (event_type=="social"){
+            event_sub = "";
+        }
+        Long event_school_L = Long.parseLong( eventJsonAsMap.get("school_id"));
+
+        // Store to the Datastore
+        Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
+        KeyFactory keyFactory = datastore.newKeyFactory().setKind("Event");
+        LatLng position = LatLng.of(event_Lat_D,event_Lng_D);
+        FullEntity eventEntity =
+           Entity.newBuilder(keyFactory.newKey())
+               .set("name", event_name)
+               .set("description", event_desc)
+               .set("location_name", event_loc)
+               .set("date", event_time)
+               .set("event_type", event_type)
+               .set("subject", event_sub)
+               .set("position", position)
+               .set("school_id", event_school_L)
+               .build();
+        datastore.put(eventEntity);
+    }
+
+    private boolean checkAllEventFieldsPresent(Map<String, String> eventJson){
```

Comment:
Can you put this fields in like a list or something?

Maybe you can create class variable

`private static String[] eventFields = {"name", ...}`;


 So that you can do something like


```
for (String field : eventFields) {
  if (!eventJson.containsKey(field)) {
    return false;
  }
}

return true;
```

I think this is a much cleaner alternative